docs/connect-extracting-data-with-mapStateToProps.md

```diff
@@ -0,0 +1,232 @@
+---
+id: connect-extracting-data-with-mapStateToProps
+title: Connect: Extracting Data with `mapStateToProps`
+hide_title: true
+sidebar_label: Connect: Extracting Data with `mapStateToProps`
+---
+
+# Connect: Extracting Data with `mapStateToProps`
+As the first argument passed in to `connect`, `mapStateToProps` is used for selecting the part of the data from the store that the connected component needs. It’s frequently referred to as just `mapState` for short.
+
+- It is called every time the store state changes.
+- It receives the entire store state, and should return an object of data this component needs.
+
+
+## Defining `mapStateToProps`
+
+`mapStateToProps` should be defined as a function:
+
+```jsx
+function mapStateToProps(state[, ownProps])
```

Comment:
What's the TS/Flow syntax for optional arguments?  I've seen the square-bracket syntax used for years with other languages and such, but wondering how helpful it is and how many people recognize that.

Reply:
In FlowType it would look like:
```
/* @flow */
type State = Object; // type for state
type Props = Object; // type of ownProps
type StateProps = Object; // type of StateProps

type mapStateToProps = (state: State, ownProps?: Props) => StateProps
```